Source/WebKit/UIProcess/ios/WKVideoView.mm

```diff
@@ -33,6 +33,23 @@ + (Class)layerClass
     return [CALayer class];
 }
 
+- (id)initWithFrame:(CGRect)frame playerView:(WebAVPlayerLayerView *)playerView
+{
+    self = [super initWithFrame:frame];
+    if (!self)
+        return nil;
+
+    [self addSubview:playerView];
+
+    return self;
+}
+
++ (CALayer *)playerLayerFromView:(UIView *)view
+{
+    ASSERT([view isKindOfClass:WKVideoView.class]);
+    return view.subviews.firstObject.layer;
+}
+
```

Comment:
Making this a class method is super weird. `-isKindOfClass:` is not expensive; in the 99% case it'll be a pointer-equality check. We shouldn't be micro-optimizing to avoid it.

Also, this is still fragile, though less so. Just add a `@property (retain, nonatomic) WebAVPlayerLayerView *playerLayerView`, assign it in the -init method, and pull the layer directly out of it.